### PR TITLE
[MINI-5532] Enable permission settings on mini app top bar

### PR DESCRIPTION
**Pull request by shohiebsense**
# Description
MiniApp Display -> Add permission settings icon to topbar -> navigate to permission setting
Remove permission settings in the Settings page.

## Links
[MINI-5532](https://jira.rakuten-it.com/jira/browse/MINI-5532)

## Video (Obsolete)
https://rak.box.com/s/8bxto2e33m8gcdny9ql1qcbfktstwz1s 

## Fix Video
https://rak.box.com/s/psvoqf9vhqpl3198p5o6w1m8uw1jh4bl

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by laubengaier — COMMENTED** [review]
@shohiebsense can you add some screenshots or a short video (using box) to show the result?

**Comment by shohiebsense** [comment]
> @shohiebsense can you add some screenshots or a short video (using box) to show the result?

Alright updated on the description @laubengaier

**Review by laubengaier — APPROVED** [review]
lgtm 👍  just ui could be prettier but we can deal with that later
ios settings screen as a reference https://rak.box.com/s/b5lohut3i1avtud9v3mfgbuppcadyhyk

**Review by laubengaier — APPROVED** [review]

**Review by rleojoseph — CHANGES_REQUESTED** [review]
I guess we should not let the user to toggle on/off settings for the permissions that is set as mandatory.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapptabs/fragments/SettingsFragment.kt [hunk 1]
```diff
@@ -244,7 +255,7 @@ class SettingsFragment : BaseFragment() {
         binding.switchPreviewMode.isChecked = settings.isPreviewMode
         binding.switchSignatureVerification.isChecked = settings.requireSignatureVerification
         binding.switchProdVersion.isChecked = settings.isProdVersionEnabled
-        if(BuildConfig.BUILD_TYPE == BuildVariant.RELEASE.value && !AppSettings.instance.isSettingSaved){
+        if (BuildConfig.BUILD_TYPE == BuildVariant.RELEASE.value && !AppSettings.instance.isSettingSaved) {
             switchProdVersion.isChecked = true
```
Comment (munir-rakuten) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapptabs/fragments/SettingsFragment.kt:259]: i think you have to add `bindings` here 
```
bindings.switchProdVersion
```

**Comment by shohiebsense** [comment]
afaik the case is moving the App's Permission Settings inside the Settings Page, it lists the mandatory permissions included to a MiniApp instead. 
Doesn't make it untestable for QA purposes for mandatory permissions? @rleojoseph cc: @laubengaier.  

Or do you want to keep the permission settings from the MiniApp displaying only optional permissions available while permission settings from the Settings Page display all the permissions...?

**Comment by munir-rakuten** [comment]
> I guess we should not let the user to toggle on/off settings for the permissions that is set as mandatory.

but for qa purpose, isn't it good to be able to change, they can check without reinstalling the app if require permission is missing.

**Comment by shohiebsense** [comment]
Or instead just removing completely and follows the iOS one like https://rak.app.box.com/file/1050932860895?s=b5lohut3i1avtud9v3mfgbuppcadyhyk no problem

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapptabs/fragments/SettingsFragment.kt (hunk 1 above)
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapptabs/fragments/SettingsFragment.kt]: aa thanks for the heads up!

**Comment by rleojoseph** [comment]
> Or instead just removing completely and follows the iOS one like https://rak.app.box.com/file/1050932860895?s=b5lohut3i1avtud9v3mfgbuppcadyhyk no problem

@shohiebsense, we should not remove it. We shouldn't provide a toggle on/off feature for the required permissions, just like iOS. 
@munir-rakuten, but we shouldn't let them to change the required permission in the first place right? So I think its not requried.

**Comment by shohiebsense** [comment]
@rleojoseph ios one currently doesn't have one, or it is removed...

https://rak.box.com/s/cu5s55futfy660kjqi7r6ov5d0xvd9li

**Comment by rleojoseph** [comment]
> Or instead just removing completely and follows the iOS one like https://rak.app.box.com/file/1050932860895?s=b5lohut3i1avtud9v3mfgbuppcadyhyk no problem

@shohiebsense , Permission Setting is moved to Mini App page now. You need to launch the miniapp to see/change the settings

**Comment by shohiebsense** [comment]
Updated the Permission Settings with only the optional permissions toggle once accepted. @rleojoseph

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**